### PR TITLE
[Bug](cast) Fix bug for cast function between datetimev2 and string

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Fix bug for cast function between datetimev2 and string

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/cast_function/test_cast_function.groovy [hunk 1]
```diff
@@ -23,6 +23,42 @@ suite("test_cast_function") {
     qt_sql """ select cast (NULL AS CHAR(1)); """
     qt_sql """ select cast ('20190101' AS CHAR(2)); """
 
-    qt_sql_null_cast_bitmap """ select cast (case when BITMAP_EMPTY() is NULL then null else null end as bitmap) is NULL; """
```
Comment (yiguolei) [inline regression-test/suites/query_p0/sql_functions/cast_function/test_cast_function.groovy:26]: why delete this case?

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**